src/patient-registration/patient-registration.component.tsx

```diff
@@ -83,22 +90,99 @@ export const PatientRegistration: React.FC = () => {
     return () => abortController.abort();
   }, []);
 
+  useEffect(() => {
+    if (existingPatient) {
+      patientUuidMap['patientUuid'] = existingPatient.id;
+      // set names
+      if (existingPatient.name) {
+        existingPatient.name.forEach((name, index) => {
+          if (index === 0) {
+            patientUuidMap['preferredNameUuid'] = name.id;
+            initialFormValues.givenName = name.given[0];
+            initialFormValues.middleName = name.given[1];
+            initialFormValues.familyName = name.family;
+          }
+          if (index === 1) {
+            patientUuidMap['additionalNameUuid'] = name.id;
+            initialFormValues.addNameInLocalLanguage = true;
+            initialFormValues.additionalGivenName = name.given[0];
+            initialFormValues.additionalMiddleName = name.given[1];
+            initialFormValues.additionalFamilyName = name.family;
+          }
+        });
+      } else {
+        initialFormValues.unidentifiedPatient = true;
```

Comment:
Does `unidentifiedPatient` really have no name? I mean I remember we did a demo and `unidentifiedPatient` had all 3 names set to unknown.

Reply:
This is correct, an `unidentifiedPatient` should not have a name, but because of the limitation of our data model, we end up having dummy values like. "UNKNOWN". I surely should look into this.